src/Dumplie/Application/Command/Customer/AddToCart.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Command\Customer;
+
+final class AddToCart
+{
+    /**
+     * @var string
+     */
+    private $sku;
+
+    /**
+     * @var int
+     */
+    private $quantity;
+
+    /**
+     * @var string
+     */
+    private $cartId;
+
+    /**
+     * @param string $sku
+     * @param int    $quantity
+     * @param string $cartId
+     */
+    public function __construct(string $sku, int $quantity, string $cartId)
+    {
+        $this->sku = $sku;
+        $this->quantity = $quantity;
+        $this->cartId = $cartId;
```

Comment:
Yeah I prefer to construct all model related stuff in application layer, so primitive type here are ok.
@norzechowicz with such approach we can just make fields public in DTO, and getters will not be needed then :)
But IMO it is not bad to "fail fast" when we know that for execution of given command some fields are required (are in constructor) we can throw exception if those are empty, so iclient code fail during construction of command and not during execution/handling of that command - it just more developer friendly ;)